### DISCUSSION TITLE
Identifier is abstract.

### DISCUSSION THREAD

**Discussion by AndreyEba** [FabricMC/fabric-api#4120] [Mod Dev Support] [answered]
### Description

I tried to create my own item and the Identifier interface gave an error that it does not accept values, but on the site docs.fabricmc.net says that you need to enter values (Mod ID and Item Name). 

package mc.andrelin.mod.realismod.item;

import mc.andrelin.mod.realismod.Realismod;
import net.fabricmc.fabric.api.item.v1.FabricItem;
import net.minecraft.core.Registry;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.Rarity;
import org.intellij.lang.annotations.Identifier;
//import org.intellij.lang.annotations.Identifier;

import static net.minecraft.core.registries.Registries.ITEM;

public class adminBanStick{
    //public FabricItem.Settings
    public static Item register(Item item, String id) {
        // Create the identifier for the item.
        //Identifier itemID = Identifier.of(Realismod.MOD_ID, id);

        // Register the item.
        Item registeredItem = Registry.register(ITEM,new Identifier(Realismod.MOD_ID, id), id, item);

        // Return the registered item!
        return registeredItem;
    }
    public static final Item AdminBanStick = register(
            new Item(new Item.Properties().rarity(Rarity.EPIC)),
            "AdminBanStick"
    );
    public static void registerItems() {

    }
}

Error:

C:\Users\KlepN\realismmod\src\main\java\mc\andrelin\mod\realismod\item\adminBanStick.java:21: error: Identifier is abstract; cannot be instantiated
        Item registeredItem = Registry.register(ITEM,new Identifier(Realismod.MOD_ID, id), id, item);
                                                     ^

**Comment by AndreyEba** [comment]
I've solved the problem
This was because I was using official mappings instead of Yarn mappings

**Comment by its-miroma** [comment, marked as answer]
Just to correct the record here: the intended approach is to use `Identifier.fromNamespaceAndPath` (mojmap) or `Identifier.of` (yarn). See FabricMC/fabric-docs#491
